oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java

```diff
@@ -290,8 +296,14 @@ static String getAsyncLaneName(String indexPath, NodeState indexState) {
     }
 
     private void releaseCheckpoint() {
-        nodeStore.release(indexerInfo.checkpoint);
-        log.info("Released the referred checkpoint [{}]", indexerInfo.checkpoint);
+        if (!preserveCheckpoint) {
```

Comment:
Nit: I would use "if (preserveCheckpoint)" and then swap the "then" and "else" parts.

Reply:
done